test/multipart/test_encoder.py

```diff
@@ -0,0 +1,287 @@
+        m = MultipartEncoder([("field", s.decode("utf-8"))])
+        assert m.read() is not None
+
+    def test_regression_1(self) -> None:
+        """Ensure issue #31 doesn't ever happen again."""
+        fields: dict[str, str | tuple[str, typing.BinaryIO]] = {"test": "t" * 100}
+
+        for x in range(30):
+            fields["f%d" % x] = ("test", open(__file__, "rb"))
+
+        m = MultipartEncoder(fields=fields)
+        total_size = len(m)
+
+        blocksize = 8192
+        read_so_far = 0
+
+        while True:
+            data = m.read(blocksize)
+            if not data:
+                break
+            read_so_far += len(data)
+
+        assert read_so_far == total_size
+
+        for k, v in fields.items():
+            if k != "test" and hasattr(v[1], "close"):
+                v[1].close()
+
+    def test_regression_2(self) -> None:
+        """Ensure issue #31 doesn't ever happen again."""
+        fields = {"test": "t" * 8100}
+
+        m = MultipartEncoder(fields=fields)
+        total_size = len(m)
+
+        blocksize = 8192
+        read_so_far = 0
+
+        while True:
+            data = m.read(blocksize)
+            if not data:
+                break
+            read_so_far += len(data)
+
+        assert read_so_far == total_size
+
+    def test_handles_empty_unicode_values(self) -> None:
+        """Verify that the Encoder can handle empty unicode strings.
+
+        See https://github.com/requests/toolbelt/issues/46 for
+        more context.
+        """
+        fields = [(b"test".decode("utf-8"), b"".decode("utf-8"))]
+        m = MultipartEncoder(fields=fields)
+        assert len(m.read()) > 0
+
+    def test_accepts_custom_content_type(self) -> None:
+        """Verify that the Encoder handles custom content-types.
+
+        See https://github.com/requests/toolbelt/issues/52
+        """
+        fields = [
+            (
+                b"test".decode("utf-8"),
+                (
+                    b"filename".decode("utf-8"),
+                    b"filecontent",
+                    b"application/json".decode("utf-8"),
+                ),
+            )
+        ]
+        m = MultipartEncoder(fields=fields)
+        output = m.read().decode("utf-8")
+        assert output.index("Content-Type: application/json\r\n") > 0
+
+    def test_accepts_custom_content_type_as_bytes(self) -> None:
+        """Verify that the Encoder handles custom content-types which
+        are bytes.
+
+        See https://github.com/requests/toolbelt/issues/52
+        """
+        fields = [
+            (
+                b"test".decode("utf-8"),
+                (
+                    b"filename".decode("utf-8"),
+                    b"filecontent",
+                    b"application/json",
+                ),
+            )
+        ]
+        m = MultipartEncoder(fields=fields)
+        output = m.read().decode("utf-8")
+        assert output.index("Content-Type: application/json\r\n") > 0
```

Comment:
This test is new, to add coverage for a couple of lines in the multipart encoder:

```python
if isinstance(file_type, bytes):
    file_type = file_type.decode("utf-8")
```